.github/mergeable.yml

```diff
@@ -0,0 +1,269 @@
+          title: 'Mergeable found some errors!'
+          summary: |
+            ### Status: {{toUpperCase validationStatus}}
+            Some or All of the validators have returned 'error' status, please check below for details
+            Here are some stats of the run: \n {{validationCount}} validations were ran.
+            {{passCount}} ***PASSED***
+            {{failCount}} ***FAILED***
+            {{errorCount}} ***ERRORED***
+          text: "{{#each validationSuites}}
+              #### {{{statusIcon status}}} Validator: {{toUpperCase name}}
+              Status {{toUpperCase status}}
+              {{#each validations }} * {{{statusIcon status}}} ***{{{ description }}}***
+                     Input : {{{details.input}}}
+                     Settings : {{{displaySettings details.settings}}}
+                     {{#if details.error}}
+                     Error : {{{details.error}}}
+                     {{/if}}
+                     {{/each}}
+              {{/each}}"
+
+  # Remove label 'status: needs review' once PR is approved.
+  - when: 'pull_request_review.*'
+    name: 'Status check'
+    validate:
+      - do: payload
+        review:
+          state:
+            must_include:
+              regex: 'approved'
+    pass:
+      - do: labels
+        delete: 'status: needs review'
+      - do: checks
+        status: neutral
+
+  # Add label 'status: needs work' if reviewer requested changes.
+  - when: 'pull_request_review.*'
+    name: 'Status check'
+    filter:
+      - do: payload
+        review:
+          state:
+            must_include:
+              regex: 'changes_requested'
+    validate:
+      - do: label
+        must_include:
+          regex: 'status: needs review'
+    pass:
+      - do: labels
+        add: 'status: needs work'
+        delete: 'status: needs review'
+      - do: comment
+        payload:
+          body: 'As the reviewer has requested changes, we have added "status: needs work" and have removed "status: needs review" changes.'
+
+  # Let's add "maintainers" team as reviewer once, 'optional: review requested' label is added.
+  - when: pull_request.*
+    name: 'Request mainatainer for review'
+    filter:
+      - do: payload
+          review:
+            state:
+              must_include:
+                regex: 'review_requested'
+    validate:
+      - do: label
+        begins_with:
+          match: 'optional: review_requested'
+    pass:
+      - do: request_review
+        teams: ['maintainers']
+      - do: comment
+          payload:
+            body: 'As the @author has requested review from maintainers, we have added @goalgorilla/maintainers reviewers.'
+
+  # We want to make sure that the cherry-pick was completed after the PR was merged.
+  # This check will run after 24 hours of PR merging.
+  # see: https://mergeable.readthedocs.io/en/latest/recipes.html?highlight=stale#check-stale-pr-and-issues
+  - when: schedule.repository
+    name: 'Check backport'
```

Comment:
This is a cool idea, curious how this will pan out though.

Imagine the label `status:needs backport` isn't there. There is no way for us to determine if a backport label should be there or not because mergable is not able to determine, based on a milestone, if something needs to be backported. 

So if they forget that label, this will already not pass.
Also when you only add the label at a later stage, probably the teams will not take a look at it anymore as it was merged.

Reply:
The idea is to check if the PR contains backport label and then verify if cherry-pick label is added or missed. This reminds developers if they have forgotten to remove the label or cherry-pick itself. But the idea is of course debatable.

Reply:
I have changed the logic please re-review.